### PR TITLE
feat: Implement fiveg_rfsim interface provider side

### PR DESCRIPTION
**Pull request by gatici**
# Description

Implement the provider side of `fiveg_rfsim` interface: https://github.com/canonical/charm-relation-interfaces/pull/175

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -87,14 +92,14 @@ def __init__(self, *args):
             ports=[
                 ServicePort(name="f1c", port=38472, protocol="SCTP"),
                 ServicePort(name="f1u", port=self._charm_config.f1_port, protocol="UDP"),
-                ServicePort(name="rfsim", port=4043, protocol="TCP"),
+                ServicePort(name="rfsim", port=RFSIM_PORT, protocol="TCP"),
```
Comment (gruyaume) [inline src/charm.py:95]: It seems some of the changes here conflict with changes in #24 let's make sure you two align on the right approach

**Review by gruyaume — APPROVED** [review]
Overall this looks good, let's just make sure you align with @Gmerold on the changes to the ports used.

File: src/charm.py (hunk 1 above)
Comment (gatici) [reply src/charm.py]: Service definition for RFSIM is not required. This line is removed.

File: src/charm.py [hunk 2]
```diff
@@ -93,7 +98,7 @@ def __init__(self, *args):
         self.framework.observe(self.on.update_status, self._configure)
         self.framework.observe(self.on.config_changed, self._configure)
         self.framework.observe(self.on.du_pebble_ready, self._configure)
-        self.framework.observe(self._f1_requirer.on.fiveg_f1_provider_available, self._configure)
```
Comment (Gmerold) [inline src/charm.py:96]: Are you sure we wanna remove this?

File: src/charm.py [hunk 3]
```diff
@@ -238,6 +244,46 @@ def _relation_created(self, relation_name: str) -> bool:
...
+    def _set_fiveg_rfsim_relation_data(self) -> None:
+        """Set rfsim information for the fiveg_rfsim relation."""
+        if not self.unit.is_leader():
+            return
+        if not self._relation_created(RFSIM_RELATION_NAME):
+            return
+        if not self._du_service_is_running():
+            return
+        if not _get_pod_ip():
+            return
```
Comment (Gmerold) [inline src/charm.py:256]: Do we need this check? `_get_rfsim_address` will validate this anyway.

**Review by Gmerold — CHANGES_REQUESTED** [review]

**Review by Gmerold — APPROVED** [review]

**Pull request merged**